File: sharmaji/d2.py
import sys
import bisect

def solve():
    data = sys.stdin.buffer.read().split()
    it = iter(data)
    array_length = int(next(it))
    query_count = int(next(it))
    prefix_sum = [0] * (array_length + 1)
    for i in range(1, array_length + 1):
        frequency = int(next(it))
        prefix_sum[i] = prefix_sum[i - 1] + frequency
    sequence_length = prefix_sum[array_length]
    out = []
    for _ in range(query_count):
        target_index = int(next(it))
        position = (target_index - 1) % sequence_length + 1
        answer = bisect.bisect_left(prefix_sum, position)
        out.append(str(answer))
    sys.stdout.write('\n'.join(out))

if __name__ == '__main__':
    solve()

File: sharmaji/d2.cpp
#include <bits/stdc++.h>
using namespace std;
int main() {
    ios::sync_with_stdio(false);
    cin.tie(0);
    int arrayLength, queryCount;
    cin >> arrayLength >> queryCount;
    vector<long long> prefixSum(arrayLength + 1);
    for (int i = 1; i <= arrayLength; ++i) {
        int frequency;
        cin >> frequency;
        prefixSum[i] = prefixSum[i - 1] + frequency;
    }
    long long sequenceLength = prefixSum[arrayLength];
    while (queryCount--) {
        long long targetIndex;
        cin >> targetIndex;
        long long position = (targetIndex - 1) % sequenceLength + 1;
        int answer = lower_bound(prefixSum.begin() + 1, prefixSum.end(), position) - prefixSum.begin();
        cout << answer << '\n';
    }
    return 0;
}